sdk/eventgrid/Azure.Messaging.EventGrid/src/autorest.md

```diff
@@ -14,56 +51,29 @@ directive:
       {
         $[path]["x-namespace"] = namespace;
       }
-      $[path]["x-csharp-usage"] = "model,output,converter";
+      if (path.endsWith("EventData") || ($[path]["x-ms-client-name"] && $[path]["x-ms-client-name"].endsWith("EventData")))
+      {
+        $[path]["x-csharp-usage"] = "model,output,converter";
+      }
+      else 
+      {
+        $[path]["x-csharp-usage"] = "model,output";
+      }
       $[path]["x-csharp-formats"] = "json";
       if (path.includes("WebAppServicePlanUpdatedEventData"))
       {
           $[path]["properties"]["sku"]["x-namespace"] = namespace;
-          $[path]["properties"]["sku"]["x-csharp-usage"] = "model,output,converter";
+          $[path]["properties"]["sku"]["x-csharp-usage"] = "model,output";
```

Comment:
`model,output`and x-csharp-formats should be applied recursively. Can you try removing them from nested models?